specification/sbi_atee.adoc

```diff
@@ -13,6 +44,936 @@ running a VMM).
 The following operations must be implemented by the TSM ( _Formal input 
 parameters and return codes are TBD_ ).
 
+== Background
+
+Virtualization platforms are typically comprised of several components including
+platform firmware, host OS, VMM, and the actual payloads that run on them (typically
+in a VM). This model is well established, but the downside is that most platform
+components are in the TCB. This aspect is ill-suited for confidential compute
+workloads (Trusted Execution Environments) that strive to minimize the TCB footprint.
+
+We seek to alleviate this concern by introducing the notion of the a trusted broker for
+TEEs known as the TEE Security Manger (TSM). The idea is that the TSM has a minimal possible
+footprint, and acts as a trusted intermediary between the the untrusted platform components,
+and the TEE workloads that run on it. This is accomplished by having a clear seperation between
+the non-confidential and confidential portions of the workloads running on the platform. The exact
+hardware and software mechanisms used to guarantee the isolation are platform dependent, but
+confidential workloads have absolute execution, data integrity, and confidentiality protections
+from amy component in the non-confidential world. The responsibility of the TSM is to enforce 
+the guarantees accorded to TEE workloads. Given its minimal footprint, most of the functionality
+for actually running the payloads, like memory allocation, scheduling, interrupt handling, 
+virtual devices is left to the OS/VMM (which are outside the TCB).
+
+In this scheme, most resources like memory, (V)CPUs start off as traditional untrusted resources
+in the non-confidential world, and are transtioned to their confidential analogue via the TSM.
+Once the conversion process is complete, a confidential-resource cannot be reverted to the 
+non-confidential world unless it's is no longer assigned to a TVM.
+
+The TEE address space can be comprised of confidential and non-confidential regions. The former includes
+both measured that are part of the TVM payload, and confidential zero-pages that can be mapped in by the
+VMM on demand. The non-confidential regions include memory that can be shared with the VMM or to emulate
+MMIO.
+
+=== Operational model
+
+Executing confidential workloads in a TEE requires a sequence of one or more of the steps detailed below.
+We'll assume that these steps are performed by an untrusted entity like the OS/VMM in conjunction with the
+TSM.
+
+. Platform TSM detection and capability enumeration
+. Conversion of non-confidential memory to confidential memory
+. Trusted VM (TVM) creation
+. Donating confidential memory to the TSM for TVM page management
+. Defining confidential memory regions
+. Defining MMIO regions
+. Defining non-confidential shared memory regions
+. Adding VCPUs to TVMs
+. Mapping code and data payloads for the TVM to confidential-memory regions
+. Mapping demand-zero confidential memory regions
```

Comment:
should this one be here - it is listed again below "Mapping of confidential zero pages on demand"